File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.java
class Solution {
    public String removeOuterParentheses(String s) {
        int open=0;
        String ans="";
        for(char ch:s.toCharArray()){
            if(ch==')')open--;
            if(open!=0)ans+=ch;
            if(ch=='(')open++;
        }
        return ans;
    }
}

File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.cpp
class Solution {
public:
    string removeOuterParentheses(string s) {
        string result;   // To store the final result
        int balance = 0; // To keep track of the balance of parentheses

        for (int i = 0; i < s.size(); i++) {
            if (s[i] == '(') {
                // If balance is greater than 0, it means this '(' is not an
                // outermost parenthesis
                if (balance > 0) {
                    result += s[i]; 
                }
                balance++;
            } else {
                balance--; 
                // If balance is greater than 0, it means this ')' is not an
                // outermost parenthesis
                if (balance > 0) {
                    result += s[i]; 
                }
            }
        }
        return result;
    }
};